File: root/common/src/main/java/org/sc/common/rest/MaintenanceResponse.java
package org.sc.common.rest;

import java.util.Collections;
import java.util.List;

public class MaintenanceResponse extends RESTResponse {
    List<Maintenance> maintenanceList;

    public MaintenanceResponse(){}

    public MaintenanceResponse(List<Maintenance> maintenanceList) {
        super(Status.OK, Collections.emptySet());
        this.maintenanceList = maintenanceList;
    }

    public List<Maintenance> getMaintenanceList() {
        return maintenanceList;
    }

}

File: root/common/src/main/java/org/sc/common/rest/response/MaintenanceResponse.kt
package org.sc.common.rest.response

import org.sc.common.rest.MaintenanceDto
import org.sc.common.rest.Status

data class MaintenanceResponse (val status: Status,
                                val messages: Set<String>,
                                val maintenanceResponse: List<MaintenanceDto>,
) : RESTResponse()